drivers/hmis/lib/tasks/setup.rake

```diff
@@ -85,3 +85,39 @@ task seed_e2e: [:environment, 'log:info_to_stdout'] do
   # Find or create ACL
   Hmis::AccessControl.where(role: role, access_group: access_group, user_group: user_group).first_or_create!
 end
+
+# rake driver:hmis:generate_custom_data_elements           # save CustomDataElements for all forms
+# rake driver:hmis:generate_custom_data_elements[SERVICE]  # save CustomDataElements for one role
+desc 'Generate Custom Data Elements by introspecting Form Definitions'
+task :generate_custom_data_elements, [:role] => [:environment, 'log:info_to_stdout'] do |_task, args|
+  role = args.role
+  custom_data_element_definitions = []
+  seen_keys = Set.new
+
+  definition_scope = Hmis::Form::Definition.non_static
+  definition_scope = definition_scope.with_role(role) if role.present?
+  puts "#{definition_scope.size} forms to process: #{definition_scope.map(&:identifier).join(', ')}"
+
+  definition_scope.order(:id).each do |definition|
```

Comment:
this should really only look at the latest version of each form identifier, but figured its not worth it now since this logic is going to move out of a rake task and into the form editor tooling